src/app/guards/auth.guard.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable } from "@angular/core";
+import { CanActivate, Router } from "@angular/router";
+import { Observable, tap } from "rxjs";
+
+import { AuthService } from "../services/auth.service";
+
+@Injectable({
+	providedIn: "root",
+})
+export class AuthGuard implements CanActivate {
+	constructor(
+		private authService: AuthService,
+		private router: Router,
```

Comment:
```suggestion
		private readonly authService: AuthService,
		private readonly router: Router,
```
best practice for injecting services
please add readonly in each injecting constructors

Reply:
also there is a newer approach to inject services. You can use it that way, it's modern
private readonly authService = inject(AuthService);